File: Semestrul 4/SDI/lab08/TriathlonServices/ITriathlonServices.cs
using Triathlon.Model;

namespace Triathlon.Services
{
    public interface ITriathlonServices
    {
        RefereeDiscipline Login(Referee referee, ITriathlonObserver client);
        void Logout(Referee referee, ITriathlonObserver client);
        List<Result> GetResultsByEvent(long eventId);
        void UpdateResult(Result result);
        int GetTotalPointsForParticipant(long participantId, long eventId);
    }
}

File: Semestrul 4/SDI/lab11/java/TriathlonServices/src/main/java/triathlon/services/ITriathlonServices.java
package triathlon.services;

import java.util.List;
import triathlon.model.*;

public interface ITriathlonServices {
    RefereeDiscipline login(Referee referee, ITriathlonObserver client) throws TriathlonException;
    void logout(Referee referee, ITriathlonObserver client) throws TriathlonException;
    List<Result> getResultsByEvent(Long eventId);
    void updateResult(Result result);
    int getTotalPointsForParticipant(Long participantId, Long eventId);
}
